### PR TITLE
add deploy external resource with okteto deploy cmd

### LINKED ISSUE #3267: Deploy external resource with `okteto deploy` command
As a developer I want to be able to deploy the external resources I have defined in my okteto manifest, so I can use that information in the UI to have a full view of my application.

The idea is that `okteto deploy` command deploys all the external resources with the label `deployed-by`

## Acceptance criteria

- [ ] okteto deploy must deploy the external resources defined in the okteto manifest
- [ ] it should throw a warning if the cluster is not a cluster managed by okteto
- [ ] When creating the new external resource item the following fields must be created
  - [ ] `deployed-by` annotation could be present
  - [ ] `last-updated` annotation must be present
  - [ ] `id` is an Autogenerated UUID
  - [ ] `name` must be present
  - [ ] `creationTime` is present
  - [ ] `Endpoints` must present
  - [ ] `Notes` is present
    - [ ] `path` is present
    - [ ] `markdown` is present

### PR DESCRIPTION
**Pull request by AdrianPedriza**
# Proposed changes

- refactor to avoid circular imports by extracting from model the necessary constants to the `constants` package 
- added handling with k8s inside the `externalresource` package
- added test covering features

Fixes #3267 

with this PR it is intended to be able to deploy a CRD external resource in k8s once the necessary information is read from the manifest. This will only be possible in clusters managed by okteto.

### CODE REVIEW

File: pkg/externalresource/control.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+func (c *K8sControl) Deploy(ctx context.Context, name, ns string, externalInfo *ExternalResource) error {
+	k8sclient, err := c.ClientProvider(*c.Cfg)
+	if err != nil {
+		return fmt.Errorf("error creating external CRD client: %s", err.Error())
+	}
+
+	externalResourceCRD := translate(name, externalInfo)
+
+	old, err := k8sclient.ExternalResources(ns).Get(ctx, externalResourceCRD.Name, metav1.GetOptions{})
```
Comment (jLopezbarb) [inline pkg/externalresource/control.go:33]: Have we tested what happens on a redeploy. Do we really update the existing external resources?

File: cmd/deploy/deploy.go [hunk 2]
```diff
@@ -498,6 +500,20 @@ func (dc *DeployCommand) deploy(ctx context.Context, opts *Options) error {
 		oktetoLog.SetStage("")
 	}
 
+	// deploy externals if any
+	if opts.Manifest.External != nil {
```
Comment (jLopezbarb) [inline cmd/deploy/deploy.go:504]: Can we add a unit test covering this scenario?

File: cmd/deploy/deploy.go [hunk 3]
```diff
@@ -287,6 +283,17 @@ func (dc *DeployCommand) RunDeploy(ctx context.Context, deployOptions *Options)
 		oktetoLog.Infof("could not create temporal kubeconfig %s", err)
 		return err
 	}
+
+	_, proxyConfig, err := dc.K8sClientProvider.Provide(kconfig.Get([]string{dc.TempKubeconfigFile}))
```
Comment (jLopezbarb) [inline cmd/deploy/deploy.go:287]: I think you can do `dc.Kubeconfig.Read` to get the cfg, that way we don't have to open and read a file

**Comment by jLopezbarb** [comment]
Unit test are failing, can you update them

**Review by jLopezbarb — APPROVED** [review]

**Comment by jLopezbarb** [comment]
Tested with the following manifest and is throwing a panic:
```yaml
deploy:
  - echo hola
external:
  local:
    endpoints:
    - url: https://localhost:8080
      name: local
    notes: a.txt
```
Also I should be able to deploy a manifest without notes

File: pkg/externalresource/control.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+func (c *K8sControl) Deploy(ctx context.Context, name, ns string, externalInfo *ExternalResource) error {
+	k8sclient, err := c.ClientProvider(c.Cfg)
+	if err != nil {
+		return fmt.Errorf("error creating external CRD client: %s", err.Error())
+	}
+
+	externalResourceCRD := translate(name, externalInfo)
+
+	old, err := k8sclient.ExternalResources(ns).Get(ctx, externalResourceCRD.Name, metav1.GetOptions{})
+	if err != nil && !oerrors.IsNotFound(err) {
```
Comment (ifbyol) [inline pkg/externalresource/control.go:34]: Shouldn't we use `k8sErrors` instead of `oerrors`?

**Pull request merged**